controllers/configentry_controller.go

```diff
@@ -0,0 +1,299 @@
+package controllers
```

Comment:
This is the generic controller that works for all config entries.

Reply:
I'd say that we should move it into sync-catalog but I think it may be used for health checks. We could rename it as something, maybe `CacheController` and move it into its own top-level package?

Reply:
Yeah, we should consolidate it I think, although I don't think this is required for this PR.